api/provider/aws/deploy_create.go

```diff
@@ -0,0 +1,131 @@
+package aws
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+func (d *DeployProvider) Create(req models.CreateDeployRequest) (*models.Deploy, error) {
+	if req.DeployName == "" {
+		return nil, fmt.Errorf("DeployName is required")
+	}
+
+	deploy, err := d.createDeploy(req.DeployName, req.Dockerrun)
+	if err != nil {
+		return deploy, err
+	}
+
+	if err := d.TagStore.Insert(models.Tag{EntityID: deploy.DeployID, EntityType: "deploy", Key: "name", Value: req.DeployName}); err != nil {
+		return deploy, err
+	}
+
+	if err := d.TagStore.Insert(models.Tag{EntityID: deploy.DeployID, EntityType: "deploy", Key: "version", Value: deploy.Version}); err != nil {
+		return deploy, err
+	}
+
+	return deploy, nil
+}
+
+func (d *DeployProvider) createDeploy(deployName string, body []byte) (*models.Deploy, error) {
```

Comment:
This helper is doing too much. The rule of thumb we've been using so far is:
* If a helper is wrapping an aws call, it should only be a single aws call
* A helper should do one "thing"
* The "Master" function (e.g. `CreateDeploy()`) is responsible for lining up all of these helper functions. 


This helper doing quite a bit: name validation, rendering, an aws call, and populating tags. 
I think [EnvironmentCreate](https://github.com/quintilesims/layer0/blob/api-refactor/api/provider/aws/environment_create.go#L49) is a good example showing how it splits out rendering `userData` and the aws api call into different functions.